tests/unit/bokeh/models/test_plots.py

```diff
@@ -192,6 +193,7 @@ def test_bad_extra_range_name(self) -> None:
         p, dep = figure(), figure()
         dep.extra_x_ranges['foo'] = Range1d()
         dep.grid.x_range_name="foo"
+        p.grid[0].js_on_change("dimension", CustomJS(code = "", args = {"toto": dep.grid[0]}))
```

Comment:
This test has gotten really overloaded. I think this last block should be split off in to a new test function like `test_only_immediate_refs`